unittests/test_fields.py

```diff
@@ -669,6 +671,22 @@ def test_delitem(self):
             KeyError, exec, "del scoped_dict['a:k4']", globals(), locals()
         )
 
+    def test_delparent_scope(self):
+        scoped_dict = fields.ScopedDict({
+            's0': {'s1': 1},
+            's0:s1': {'k0': 2}
+        })
+
+        self.assertEqual(1, scoped_dict['s0:s1'])
+        self.assertEqual(2, scoped_dict['s0:s1:k0'])
+
+        # delete the parent scope
+        del scoped_dict['s0']
```

Comment:
Check my comment in the issue. The desired behaviour is the one described there, so there is a bug in the `__delitem__`. Besides, I think there is already a unit tests for deleting a whole scope.